gradle/s3sync.gradle

```diff
@@ -72,6 +72,12 @@ def syncData(subdir) {
   while (needMoreData) {
       def remoteFiles = new XmlSlurper().parseText(new URL("https://h2o-public-test-data.s3.amazonaws.com/?marker=$lastKey").getText())
       needMoreData = remoteFiles.IsTruncated == "true"
+      // If no such file appears on the first result page skip to next page
+      if (remoteFiles.Contents.findAll{it.Key =~ subdir}.size() == 0) {
```

Comment:
This can still be improved. You should never consider the filter for the last key and skip the whole page in the next iteration regardless of content of the page.

Reply:
Okay, I see, I could skip just one item, but on the other hand it could reduce the amount of HTTP requests made if it was safe..
BTW is it possible that some item won't have a Key attribute, or Key attribute equal to empty string?

Reply:
I want to skip the whole page not just one item.

Imagine the page looks like this

```
a - accepted
b - filtered out
c - filtered out
d - filtered out
```

Your if-statement doesn't kick in because there is one accepted file. That means the last key will be "a". Which means then next page will start with "b".

Correct solution would skip the whole page.